Source/PeleLM_derive.cpp

```diff
@@ -695,3 +695,182 @@ void pelelm_dcma (const Box& bx, FArrayBox& derfab, int dcomp, int ncomp,
 #endif
     });
 }
+
+//
+//  Compute vorticity
+//
+void pelelm_mgvort (const Box& bx, FArrayBox& derfab, int dcomp, int ncomp,
+                    const FArrayBox& datfab, const Geometry& geomdata,
+                    Real time, const int* bcrec, int level)
+
+{
+    AMREX_ASSERT(derfab.box().contains(bx));
+    AMREX_ASSERT(datfab.box().contains(bx));
+
+    AMREX_D_TERM(const amrex::Real idx = geomdata.InvCellSize(0);,
+                 const amrex::Real idy = geomdata.InvCellSize(1);,
+                 const amrex::Real idz = geomdata.InvCellSize(2););
+
+#ifdef AMREX_USE_EB
+    const auto& fact = EBFactory(level);
+    const auto& flags_mf = fact.getMultiEBCellFlagFab();
+#endif
+
+    amrex::Array4<amrex::Real const> const& dat_arr = datfab.const_array();
+    amrex::Array4<amrex::Real>       const&vort_arr = derfab.array();
+
+#ifdef AMREX_USE_EB
+    const EBCellFlagFab& flags = flags_mf[mfi];
+    auto typ = flags.getType(bx);
+    if (typ == FabType::covered)
+    {
+        amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
+        {
+            vort_arr(i,j,k) = 0.0;
+        });
+    } else if (typ == FabType::singlevalued)
+    {
+       const auto& flag_fab = flags.const_array();
+       amrex::ParallelFor(bx, [=] AMREX_GPU_DEVICE (int i, int j, int k) noexcept
+       {
+          constexpr amrex::Real c0 = -1.5;
+          constexpr amrex::Real c1 = 2.0;
+          constexpr amrex::Real c2 = -0.5;
+          if (flag_fab(i,j,k).isCovered())
+          {
+             vort_arr(i,j,k) = 0.0;
+          } else {
+             amrex::Real vx = 0.0;
+             amrex::Real wx = 0.0;
+#if ( AMREX_SPACEDIM == 2 )
+             // Need to check if there are covered cells in neighbours --
+             // -- if so, use one-sided difference computation (but still quadratic)
+             if (!flag_fab(i,j,k).isConnected( 1,0,0)) {
+                vx = - (c0 * dat_arr(i  ,j,k,1)
+                      + c1 * dat_arr(i-1,j,k,1)
+                      + c2 * dat_arr(i-2,j,k,1)) * idx;
+             } else if (!flag_fab(i,j,k).isConnected(-1,0,0)) {
+                vx = (c0 * dat_arr(i  ,j,k,1)
+                    + c1 * dat_arr(i+1,j,k,1)
+                    + c2 * dat_arr(i+2,j,k,1)) * idx;
+             } else {
+                vx = 0.5 * (dat_arr(i+1,j,k,1) - dat_arr(i-1,j,k,1)) * idx;
+             }
```

Comment:
Doesn't his require that the ``i+/-2`` cells be connected?

Reply:
Indeed. I pulled that from incflo, so they must assume that 2 EB cut-cells are always 2 regular cells away from each other. We are also gonna have troubles when we include the BC stencils, but I'll admit I was thinking of letting tomorrow me think about that ...

Reply:
...tough critics in this bunch. :)